From a62b60fe7c0254f447f32086d742295c82b5e978 Mon Sep 17 00:00:00 2001
From: Mahesh Vaidya <mahesh.r.vaidya@intel.com>
Date: Wed, 16 Feb 2022 02:14:14 +0530
Subject: [PATCH] EHL DMA burst size for SPI decreased to fifolen/4

DMA burst size for SPI decreased from existing fifolen/2 to fifolen/4,
as the DMA currently isn't working with higher burst size for SPI.

Signed-off-by: Mahesh Vaidya <mahesh.r.vaidya@intel.com>
---
 drivers/spi/spi-dw-dma.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/spi/spi-dw-dma.c b/drivers/spi/spi-dw-dma.c
index a09831c62192..0dc291d506a4 100644
--- a/drivers/spi/spi-dw-dma.c
+++ b/drivers/spi/spi-dw-dma.c
@@ -39,7 +39,7 @@ static void dw_spi_dma_maxburst_init(struct dw_spi *dws)
 	u32 max_burst, def_burst;
 	int ret;
 
-	def_burst = dws->fifo_len / 2;
+	def_burst = dws->fifo_len / 4;
 
 	ret = dma_get_slave_caps(dws->rxchan, &caps);
 	if (!ret && caps.max_burst)
-- 
2.31.0

